packages/services/f-braze-adapter/src/BrazeDispatcher.js

```diff
@@ -20,22 +20,29 @@ function interceptInAppMessageClickEventsHandler (message) {
  * click events of the CTA button
  * @param message
  * @this BrazeDispatcher
+ * @return Promise
  */
 function interceptInAppMessagesHandler (message) {
-    if (message instanceof this.appboy.ab.InAppMessage) {
-        /**
-         * Always subscribe click action to second button
-         * as this is always "success" as opposed to "dismiss"
-         * as confirmed with CRM (AS)
-         */
-        this.inAppMessagesCallbacks.forEach(callback => callback(message));
-        if (message.buttons && message.buttons.length >= 2) {
-            const [, button] = message.buttons;
-            // Note that the below subscription returns an ID that could later be used to unsubscribe
-            button.subscribeToClickedEvent(() => interceptInAppMessageClickEventsHandler.bind(this)(message));
-        }
-    }
-    this.appboy.display.showInAppMessage(message);
+    return this.appboyPromise
+        .then(appboy => {
+            if (message instanceof appboy.ab.InAppMessage) {
+                /**
+                 * Always subscribe click action to second button
+                 * as this is always "success" as opposed to "dismiss"
+                 * as confirmed with CRM (AS)
+                 */
+                this.inAppMessagesCallbacks.forEach(callback => callback(message));
+                if (message.buttons && message.buttons.length >= 2) {
+                    const [, button] = message.buttons;
+                    // Note that the below subscription returns an ID that could later be used to unsubscribe
+                    button.subscribeToClickedEvent(() => interceptInAppMessageClickEventsHandler.bind(this)(message));
+                }
+            }
+            appboy.display.showInAppMessage(message);
+        })
+        .catch(error => {
+            this.logger('error', `Error handling message - ${error}`, { message, error });
```

Comment:
Where does `this.logger` come from? Is it something we need to configure in consuming apps?

Reply:
From [here](https://github.com/justeat/fozzie-components/blob/e192a3516766f5620c3d79fe051339cb57938c03/packages/services/f-braze-adapter/src/BrazeDispatcher.js#L136) - I'll message you regarding internal resources for this.